includes/class-wc-coupon.php

```diff
@@ -690,8 +697,20 @@ public function read_manual_coupon( $code, $coupon ) {
 					break;
 			}
 		}
-		$this->set_code( $code );
 		$this->set_props( $coupon );
+		$this->set_code( $code );
+		$this->set_id( 0 );
```

Comment:
I see you updated the code in `WC_Discounts` which is good. But can it potentially cause issues within `WC_Coupon_Data_Store_CPT::read` where we also check for coupon id? Similarly in legacy `WC_API_Coupons::get_coupon`.

Reply:
`true` evaluates to 1 when cast to int, so this is here to avoid it thinking coupon ID 1 exists :)